### PR TITLE
Fix/uno 489/rdf value substring search

### PR DESCRIPTION
**Pull request by zagovorichev**
Pull request summary
 
Related to : https://oat-sa.atlassian.net/browse/UNO-489

####  Description

Have in manage schema a property for "languages" with values like "Simplified Chinese from China"
"Traditional Chinese from Taiwan"

When I create a test, adding this property I search for "Simplified" 
--> I get resultst

When I search for "Chine" 
--> I don't get any results. Which we need

If the list element label consists of more than 1 word, the search for second (third, etc.) word is not working. User is able to look for a property only by the first word.

Preconditions: created list with elements, labels of which consist of more than 1 word.

#### Steps to reproduce:

Add new property via Manage Schema of a class;

Set the 'List - Single choice - Search input' or 'List - Multiple choice - Search input' as a property type to a class;

Select created list, labels of which consist of more than 1 word as 'List values';

Try to search for an item property by second word of the list element.

Actual result: property is not found.
Expected result: user is able to look for a property by second (third, etc.) word.

#### How to test
 
Describe testing process...
 
Test commands :
 
```
sudo -u www-data ./vendor/bin/phpunit tao/test/unit/models/classes/Lists/DataAccess/Repository/RdfValueCollectionRepositoryTest.php
```

### CODE REVIEW

File: models/classes/Lists/DataAccess/Repository/RdfValueCollectionRepository.php [hunk 1]
```diff
@@ -340,7 +340,7 @@ private function enrichQueryWithSubject(ValueCollectionSearchRequest $searchRequ
                     ':subject'
                 )
             )
-            ->setParameter('subject', $searchRequest->getSubject() . '%');
+            ->setParameter('subject', '%' . $searchRequest->getSubject() . '%');
```
Comment (wazelin) [inline models/classes/Lists/DataAccess/Repository/RdfValueCollectionRepository.php:343]: Such a condition without a full-text index may be very slow. Could you please compare `EXPLAIN` queries?

Also, the same should be applied to the RDS repository implementation.

Tests should be adapted as well.

**Review by wazelin — CHANGES_REQUESTED** [review]

File: models/classes/Lists/DataAccess/Repository/RdfValueCollectionRepository.php (hunk 1 above)
Comment (zagovorichev) [reply models/classes/Lists/DataAccess/Repository/RdfValueCollectionRepository.php:343]: am working on the tests part, will add you to the reviewers, after these fixes, thank you for your commentary

Comment (zagovorichev) [reply models/classes/Lists/DataAccess/Repository/RdfValueCollectionRepository.php:343]: not sure that I've understood why do you need explain here, as it is the same query with same like, but:

Query:
```
explain SELECT collection.object as collection_uri, element.id, element.subject, element.object 
  FROM statements element
    INNER JOIN statements collection ON collection.subject = element.subject 
    INNER JOIN statements property ON property.object = collection.object 
  WHERE (element.predicate = 'http://www.w3.org/2000/01/rdf-schema#label')
    AND (collection.predicate = 'http://www.w3.org/1999/02/22-rdf-syntax-ns#type')
    AND (property.subject = 'http://tao-udir.loc/tao-udir.rdf#i5f630dc167712691ec797105c10dc765')
    AND (property.predicate = 'http://www.w3.org/2000/01/rdf-schema#range')
    AND (collection.object = 'http://www.tao.lu/Ontologies/TAO.rdf#Languages')
    AND (LOWER(element.object) LIKE 'chin%')
    AND (element.subject NOT IN (''))
  ORDER BY element.id ASC LIMIT 20
```

Result with '%string%':
```
Limit  (cost=73.58..73.58 rows=1 width=122)
  ->  Sort  (cost=73.58..73.58 rows=1 width=122)
        Sort Key: element.id
        ->  Nested Loop  (cost=5.29..73.57 rows=1 width=122)
              ->  Nested Loop  (cost=4.88..65.13 rows=1 width=122)
                    ->  Index Scan using k_po on statements collection  (cost=0.41..12.45 rows=2 width=88)
                          Index Cond: (((predicate)::text = 'http://www.w3.org/1999/02/22-rdf-syntax-ns#type'::text) AND (object = 'http://www.tao.lu/Ontologies/TAO.rdf#Languages'::text))
                    ->  Bitmap Heap Scan on statements element  (cost=4.47..26.33 rows=1 width=92)
                          Recheck Cond: (((subject)::text = (collection.subject)::text) AND ((predicate)::text = 'http://www.w3.org/2000/01/rdf-schema#label'::text))
                          Filter: (((subject)::text <> ''::text) AND (lower(object) ~~ '%chin%'::text))
                          ->  Bitmap Index Scan on k_sp  (cost=0.00..4.47 rows=6 width=0)
                                Index Cond: (((subject)::text = (collection.subject)::text) AND ((predicate)::text = 'http://www.w3.org/2000/01/rdf-schema#label'::text))
              ->  Index Scan using k_po on statements property  (cost=0.41..8.43 rows=1 width=30)
                    Index Cond: (((predicate)::text = 'http://www.w3.org/2000/01/rdf-schema#range'::text) AND (object = 'http://www.tao.lu/Ontologies/TAO.rdf#Languages'::text))
                    Filter: ((subject)::text = 'http://tao-udir.loc/tao-udir.rdf#i5f630dc167712691ec797105c10dc765'::text)
```

Query with 'string%':
```
Limit  (cost=73.58..73.58 rows=1 width=122)
  ->  Sort  (cost=73.58..73.58 rows=1 width=122)
        Sort Key: element.id
        ->  Nested Loop  (cost=5.29..73.57 rows=1 width=122)
              ->  Nested Loop  (cost=4.88..65.13 rows=1 width=122)
                    ->  Index Scan using k_po on statements collection  (cost=0.41..12.45 rows=2 width=88)
                          Index Cond: (((predicate)::text = 'http://www.w3.org/1999/02/22-rdf-syntax-ns#type'::text) AND (object = 'http://www.tao.lu/Ontologies/TAO.rdf#Languages'::text))
                    ->  Bitmap Heap Scan on statements element  (cost=4.47..26.33 rows=1 width=92)
                          Recheck Cond: (((subject)::text = (collection.subject)::text) AND ((predicate)::text = 'http://www.w3.org/2000/01/rdf-schema#label'::text))
                          Filter: (((subject)::text <> ''::text) AND (lower(object) ~~ 'chin%'::text))
                          ->  Bitmap Index Scan on k_sp  (cost=0.00..4.47 rows=6 width=0)
                                Index Cond: (((subject)::text = (collection.subject)::text) AND ((predicate)::text = 'http://www.w3.org/2000/01/rdf-schema#label'::text))
              ->  Index Scan using k_po on statements property  (cost=0.41..8.43 rows=1 width=30)
                    Index Cond: (((predicate)::text = 'http://www.w3.org/2000/01/rdf-schema#range'::text) AND (object = 'http://www.tao.lu/Ontologies/TAO.rdf#Languages'::text))
                    Filter: ((subject)::text = 'http://tao-udir.loc/tao-udir.rdf#i5f630dc167712691ec797105c10dc765'::text)
```

**Comment by zagovorichev** [comment]
@wazelin @AndreyNegrov PR is ready, please check it again

**Review by AndreyNegrov — APPROVED** [review]
- [ ] New code is covered by tests (if applicable)
- [ ] Tests are running successfully (old and new ones) on my local machine (if applicable)
- [x] New code is respecting code style rules
- [x] New code is respecting best practices
- [x] New code is not subject to concurrency issues (if applicable)
- [x] Feature is working correctly on my local machine (if applicable)
- [x] Acceptance criteria are respected
- [x] Pull request title and description are meaningful

**Review by wazelin — APPROVED** [review]
- [x] New code is covered by tests (if applicable)
- [x] Tests are running successfully (old and new ones) on my local machine (if applicable)
- [x] New code is respecting code style rules
- [x] New code is respecting best practices
- [x] New code is not subject to concurrency issues (if applicable)
- [x] Feature is working correctly on my local machine (if applicable)
- [x] Acceptance criteria are respected
- [ ] Pull request title and description are meaningful ⚠️ consider changing the title to something more feature-oriented, like "full-text lookup input search"
- [x] Pull request's target is not `master`

**Pull request merged**